sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeter.java

```diff
@@ -45,21 +68,44 @@ Collection<MetricData> collectAll(
 
   @Override
   public LongCounterBuilder counterBuilder(String name) {
-    return new SdkLongCounter.Builder(meterProviderSharedState, meterSharedState, name);
+    return !isValidName(name)
+        ? MeterProvider.noop().meterBuilder(name).build().counterBuilder(name)
+        : new SdkLongCounter.Builder(meterProviderSharedState, meterSharedState, name);
   }
 
   @Override
   public LongUpDownCounterBuilder upDownCounterBuilder(String name) {
-    return new SdkLongUpDownCounter.Builder(meterProviderSharedState, meterSharedState, name);
+    return !isValidName(name)
+        ? MeterProvider.noop().meterBuilder(name).build().upDownCounterBuilder(name)
+        : new SdkLongUpDownCounter.Builder(meterProviderSharedState, meterSharedState, name);
   }
 
   @Override
   public DoubleHistogramBuilder histogramBuilder(String name) {
-    return new SdkDoubleHistogram.Builder(meterProviderSharedState, meterSharedState, name);
+    return !isValidName(name)
+        ? MeterProvider.noop().meterBuilder(name).build().histogramBuilder(name)
+        : new SdkDoubleHistogram.Builder(meterProviderSharedState, meterSharedState, name);
   }
 
   @Override
   public DoubleGaugeBuilder gaugeBuilder(String name) {
-    return new SdkDoubleGaugeBuilder(meterProviderSharedState, meterSharedState, name);
+    return !isValidName(name)
+        ? MeterProvider.noop().meterBuilder(name).build().gaugeBuilder(name)
+        : new SdkDoubleGaugeBuilder(meterProviderSharedState, meterSharedState, name);
+  }
+
+  // Visible for testing
+  static boolean isValidName(String name) {
+    if (name == null || !VALID_INSTRUMENT_NAME.matcher(name).matches()) {
```

Comment:
is it worth putting this warning in the API as well, so instrumentation authors will know that they are doing something wrong, even without an SDK in place?

Reply:
By API, do you mean the noop implementation `DefaultMeter`?

Reply:
yes